aiida_quantumespresso/calculations/neb.py

```diff
@@ -5,438 +5,316 @@
+        #spec.exit_code(
+        #    130, 'ERROR_JOB_NOT_DONE', message='The computation did not finish properly (\'JOB DONE\' not found).')
+        # TODO: check error logic and maybe use these commented-out exit codes
+        spec.exit_code(312, 'ERROR_OUTPUT_STDOUT_INCOMPLETE',
+            message='The stdout output file was incomplete.')
+        spec.exit_code(320, 'ERROR_OUTPUT_XML_READ',
+            message='The XML output file could not be read.')
+        spec.exit_code(321, 'ERROR_OUTPUT_XML_PARSE',
+            message='The XML output file could not be parsed.')
+        spec.exit_code(322, 'ERROR_OUTPUT_XML_FORMAT',
+            message='The XML output file has an unsupported format.')
+        spec.exit_code(350, 'ERROR_UNEXPECTED_PARSER_EXCEPTION',
+            message='The parser raised an unexpected exception.')
+
+    @classmethod
+    def _generate_NEBinputdata(cls, neb_parameters, settings_dict):
         """
         This methods generate the input data for the NEB part of the calculation
         """
         # I put the first-level keys as uppercase (i.e., namelist and card names)
         # and the second-level keys as lowercase
         # (deeper levels are unchanged)
-        input_params = _uppercase_dict(neb_parameters.get_dict(),
-                                       dict_name='parameters')
-        input_params = {k: _lowercase_dict(v, dict_name=k)
-                        for k, v in six.iteritems(input_params)}
-
-        # For the neb input there is no blocked keyword
-
-        # Create an empty dictionary for the compulsory namelist 'PATH'
-        # if not present
+        input_params = _uppercase_dict(neb_parameters.get_dict(), dict_name='parameters')
+        input_params = {k: _lowercase_dict(v, dict_name=k) for k, v in six.iteritems(input_params)}
+
+        # Force default values for blocked keywords. NOTE: this is different from PW/CP
+        for blocked in cls._blocked_keywords:
+            namelist = blocked[0].upper()
+            key = blocked[1].lower()
+            value = blocked[2]
+            if namelist in input_params:
+                if key in input_params[namelist]:
+                    raise InputValidationError(
+                        "You cannot specify explicitly the '{}' key in the '{}' "
+                        'namelist.'.format(key, namelist))
+            else:
+                input_params[namelist] = {}
+            input_params[namelist][key] = value
+
+        # Create an empty dictionary for the compulsory namelist 'PATH' if not present
         if 'PATH' not in input_params:
             input_params['PATH'] = {}
 
         # In case of climbing image, we need the corresponding card
-        climbing_image = False
-        if input_params['PATH'].get('ci_scheme', 'no-ci').lower()  in ['manual']:
-            climbing_image = True
+        manual_climbing_image = False
+        if input_params['PATH'].get('ci_scheme', 'no-ci').lower() in ['manual']:
```

Comment:
`in ['manual']` could be simplified to `== 'manual'`?

Reply:
Yeah